lib/Driver/Compilation.cpp

```diff
@@ -410,6 +449,7 @@ namespace driver {
       SmallVector<const Job *, 16> Dependents;
       if (Comp.getIncrementalBuildEnabled()) {
         reloadAndRemarkDeps(FinishedCmd, ReturnCode, Dependents);
+        formBatchJobsAndAddPendingJobsToTaskQueue();
```

Comment:
I had to go figure out how `reloadAndRemarkDeps` could schedule new jobs. Maybe there should be a comment around that. (My fault, not yours.)

Reply:
Fixing. Though .. there's something a little hinky here that I was kinda papering over and I'm still uncomfortable with: we _should_ pick up the extra pending jobs and requeue them in the outer loop (runTaskQueueToCompletion); I'm going to take another look and see if I can figure out why were weren't (there's a single test that fails without one of these seemingly-redundant formBatchJobsAndAddPendingJobsToTaskQueue calls).

Reply:
Oh gosh, ok, found the underlying issue here and it's both quite subtle and would have been pretty serious if I'd let it go.

(I put a couple comments on the reload path anyways, but it's not the culprit)